src/zenml/integrations/label_studio/label_studio_utils.py

```diff
@@ -0,0 +1,119 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for the Label Studio annotator integration."""
+
+import os
+from typing import Any, Dict, List, Optional, Tuple
+from urllib.parse import quote, urlparse
+
+from zenml.io import fileio
+
+
+def parse_azure_url(url: str) -> Tuple[str, str]:
+    """Converts Azure Label Studio URL to path for fileio.
+
+    Args:
+        url: Azure Label Studio URL.
+
+    Returns:
+        Tuple of (full URL, filename).
+    """
+    pth = urlparse(url).path
+    return f"az://{pth}", pth.split("/")[-1]
+
+
+def download_image(url: str, destination: str) -> None:
+    """Downloads an image using fileio.
+
+    Args:
+        url: URL of the image to download.
+        destination: Path to the destination file.
+    """
+    full_url, filename = parse_azure_url(url)
+    fileio.copy(full_url, f"{destination}/{filename}")
+
+
+def get_azure_credentials() -> Tuple[Optional[str], Optional[str]]:
+    """Returns access credentials for Azure from the environment.
+
+    Returns:
+        Tuple of (account_name, account_key).
+    """
+    account_key = os.environ.get("AZURE_STORAGE_ACCOUNT_KEY")
+    account_name = os.environ.get("AZURE_STORAGE_ACCOUNT_NAME")
+    return account_name, account_key
```

Comment:
This approach unfortunately won't work when running in e.g. Kubeflow, as these environment variables (even when set locally) won't be set when executing the steps. We should probably refactor this to also include the `authentication_secret` defined on the artifact stores.

Reply:
Same for GCS/AWS below